tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/server/QueueQueryTest.java

```diff
@@ -155,19 +156,20 @@ public void testQueueQueryOnAutoCreatedQueueWithFQQN() throws Exception {
       SimpleString addressName = SimpleString.toSimpleString(UUID.randomUUID().toString());
       SimpleString queueName = SimpleString.toSimpleString(UUID.randomUUID().toString());
       SimpleString fqqn = addressName.concat("::").concat(queueName);
-      JMSContext c = new ActiveMQConnectionFactory("vm://0").createContext();
-      c.createProducer().send(c.createQueue(fqqn.toString()), c.createMessage());
-      QueueQueryResult queueQueryResult = server.queueQuery(fqqn);
-      assertEquals(queueName, queueQueryResult.getName());
-      assertEquals(addressName, queueQueryResult.getAddress());
-      assertEquals(1, queueQueryResult.getMessageCount());
-      queueQueryResult = server.queueQuery(queueName);
-      assertEquals(queueName, queueQueryResult.getName());
-      assertEquals(addressName, queueQueryResult.getAddress());
-      assertEquals(1, queueQueryResult.getMessageCount());
-      c.createProducer().send(c.createQueue(addressName.toString()), c.createMessage());
-      assertEquals(2, server.queueQuery(fqqn).getMessageCount());
-      assertEquals(2, server.queueQuery(queueName).getMessageCount());
+      try (JMSContext c = new ActiveMQConnectionFactory("vm://0").createContext()) {
+         c.createProducer().send(c.createQueue(fqqn.toString()), c.createMessage());
+         QueueQueryResult queueQueryResult = server.queueQuery(fqqn);
+         assertEquals(queueName, queueQueryResult.getName());
+         assertEquals(addressName, queueQueryResult.getAddress());
+         Wait.assertEquals(1, () -> server.queueQuery(fqqn).getMessageCount());
+         queueQueryResult = server.queueQuery(queueName);
+         assertEquals(queueName, queueQueryResult.getName());
+         assertEquals(addressName, queueQueryResult.getAddress());
+         assertEquals(1, queueQueryResult.getMessageCount());
+         c.createProducer().send(c.createQueue(addressName.toString()), c.createMessage());
+         Wait.assertEquals(2, () -> server.queueQuery(fqqn).getMessageCount());
```

Comment:
Couldn't it be:
Wait.assertEquals(2, server.queueQuery(fqqn)::getMessageCount); 

or you actually needed to perform a query every iteration on the loop?

no biggie.. I'm going to merge it regardless... just telling you so you know.

Reply:
merged.. however I ammended this change

Reply:
The test in the amended commit still fails after a handful of iterations. The query needs to keep running so that it gets the new messageCount from the server otherwise the messageCount never changes.